moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -35,42 +35,45 @@
 /* Author: Ioan Sucan */
 
 #include <moveit/planning_pipeline/planning_pipeline.h>
+#include <moveit/planning_pipeline/planning_pipeline_middleware_handle.hpp>
 #include <moveit/robot_state/conversions.h>
 #include <moveit/collision_detection/collision_tools.h>
 #include <moveit/trajectory_processing/trajectory_tools.h>
 #include <boost/tokenizer.hpp>
 #include <boost/algorithm/string/join.hpp>
 #include <sstream>
+#include <memory>
 
 static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit.ros_planning.planning_pipeline");
 
 const std::string planning_pipeline::PlanningPipeline::DISPLAY_PATH_TOPIC = "display_planned_path";
 const std::string planning_pipeline::PlanningPipeline::MOTION_PLAN_REQUEST_TOPIC = "motion_plan_request";
 const std::string planning_pipeline::PlanningPipeline::MOTION_CONTACTS_TOPIC = "display_contacts";
 
-planning_pipeline::PlanningPipeline::PlanningPipeline(const moveit::core::RobotModelConstPtr& model,
-                                                      const std::shared_ptr<rclcpp::Node>& node,
-                                                      const std::string& parameter_namespace,
-                                                      const std::string& planner_plugin_param_name,
-                                                      const std::string& adapter_plugins_param_name)
-  : node_(node), parameter_namespace_(parameter_namespace), robot_model_(model)
+planning_pipeline::PlanningPipeline::PlanningPipeline(
+    const moveit::core::RobotModelConstPtr& model,
+    std::unique_ptr<PlanningPipeline::MiddlewareHandle> middleware_handle, const std::string& parameter_namespace,
+    const std::string& planner_plugin_param_name, const std::string& adapter_plugins_param_name)
+  : middleware_handle_(std::move(middleware_handle))
+  , parameter_namespace_(parameter_namespace)
+  , robot_model_(std::move(model))
```

Comment:
Moving the robot model will not do anything in this case because the parameter is a const pointer to a const object meaning `std::move` won't do anything. Copying the pointer is the best we can do here and therefore the `std::move` should be removed.